libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -371,57 +368,67 @@ bool AP_ExternalAHRS_VectorNav::decode(char c)
 }
 
 // decode the most recently consumed term
-// returns true if new sentence has just passed checksum test and is validated
+// returns true if new term is valid
 bool AP_ExternalAHRS_VectorNav::decode_latest_term()
 {
+    // Check the first two terms (In most cases header + reg number) that they match the sent
+    // message. If not, the response is invalid.
     switch (nmea.term_number) {
         case 0:
-            if (strcmp(nmea.term, "VNRRG") != 0) {
+            if (strncmp(nmea.term, message_to_send, nmea.term_offset) !=
+                0) {  // ignore sync byte in comparison
                 return false;
+            } else if (strncmp(nmea.term, "VNERR", nmea.term_offset) == 0) {
+                AP_HAL::panic("VectorNav received VN error response");
             }
             break;
-
-        case 1:
-            if (nmea.register_number != strtoul(nmea.term, nullptr, 10)) {
+        case 1: 
+            if (strncmp(nmea.term, message_to_send + 6,
+                        nmea.term_offset) != 0) {  // Start after "VNXXX,"
                 return false;
             }
             break;
-
-        case 2:
-            strncpy(model_name, nmea.term, sizeof(model_name));
+        case 2: 
+            if (strncmp(nmea.term, "VN-", 3) == 0) {
+                // This term is the model number
+                strncpy(model_name, nmea.term, sizeof(model_name));
+                break;
+            }
             break;
-
         default:
-            return false;
+            break;
     }
     return true;
 }
 
-void AP_ExternalAHRS_VectorNav::update_thread()
-{
+void AP_ExternalAHRS_VectorNav::initialize() {
     // Open port in the thread
     uart->begin(baudrate, 1024, 512);
 
-    // Reset and wait for module to reboot
-    // VN_100 takes 1.25 seconds
-    //nmea_printf(uart, "$VNRST");
-    //hal.scheduler->delay(3000);
+    // Pause asynchronous communications to simplify packet finding
+    hal.util->snprintf(message_to_send, sizeof(message_to_send), "$VNASY,0");
```

Comment:
```suggestion
    hal.util->snprintf(message_to_send, sizeof(message_to_send), "VNASY,0");
```
`send_command_blocking` adds the `$` for you.

Reply:
complete